codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/Writable.kt

```diff
@@ -18,14 +18,54 @@ typealias Writable = RustWriter.() -> Unit
  */
 fun writable(w: Writable): Writable = w
 
-fun writable(w: String): Writable = writable { rust(w) }
+fun writable(w: String): Writable = writable { writeInline(w) }
 
 fun Writable.isEmpty(): Boolean {
     val writer = RustWriter.root()
     this(writer)
     return writer.toString() == RustWriter.root().toString()
 }
 
+/**
+ * Helper allowing a `Iterable<Writable>` to be joined together using a `String` separator.
+ */
+fun Iterable<Writable>.join(separator: String = ", ") = join(writable(separator))
+
+/**
+ * Helper allowing a `Iterable<Writable>` to be joined together using a `Writable` separator.
+ */
+fun Iterable<Writable>.join(separator: Writable = writable { }): Writable {
+    val iter = this.iterator()
+    return writable {
+        iter.forEach { value ->
+            value()
+            if (iter.hasNext()) {
+                separator()
+            }
+        }
+    }
+}
+
+/**
+ * Helper allowing a `Sequence<Writable>` to be joined together using a `String` separator.
+ */
+fun Sequence<Writable>.join(separator: String = ", ") = asIterable().join(separator)
```

Comment:
this seems like it would be very confusing which one the compiler picked—sequence vs. iterable. One will have `,` but the other will have no separator

Reply:
whoops, yeah - this should be `writable(", ")`.